src/osd/sdl/video.cpp

```diff
@@ -109,15 +109,23 @@ void sdl_osd_interface::update(bool skip_redraw)
 //      profiler_mark(PROFILER_END);
 	}
 
-	// poll the joystick values here
-	downcast<sdl_osd_interface&>(machine().osd()).poll_inputs(machine());
-
-	check_osd_inputs(machine());
 	// if we're running, disable some parts of the debugger
 	if ((machine().debug_flags & DEBUG_FLAG_OSD_ENABLED) != 0)
 		debugger_update();
 }
 
+//============================================================
+//  input_update
+//============================================================
+
+void sdl_osd_interface::input_update(void)
+{
+	// poll the joystick values here
+	downcast<sdl_osd_interface&>(machine().osd()).process_events_buf();
+	downcast<sdl_osd_interface&>(machine().osd()).poll_inputs(machine());
```

Comment:
Why are you calling `machine().osd()` and doing a downcast here when you’re already in a `sdl_osd_interface` member?

Reply:
Indeed. I'm just copying and pasting from current source. I'll fix it.